Improve Yeelight code

shenxn opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
- Rename `ipaddr` to `host`
- Move custom services to entity services

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-asc+-review%3Aapproved

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

shenxn committed:
Rename ipaddr to ip_addr

shenxn committed:
Move custom services to entity services

shenxn committed:
Remove platform data

rytilahti commented on homeassistant/components/yeelight/__init__.py:
```diff
@@ -163,10 +160,10 @@ async def async_setup(hass: HomeAssistant, config: dict) -> bool:
     # Import manually configured devices
-    for ipaddr, device_config in config.get(DOMAIN, {}).get(CONF_DEVICES, {}).items():
-        _LOGGER.debug("Importing configured %s", ipaddr)
+    for ip_addr, device_config in config.get(DOMAIN, {}).get(CONF_DEVICES, {}).items():
```
Should it be `host` instead of `ip_addr`, assuming that hostnames are also accepted?

MartinHjelmare commented on homeassistant/components/yeelight/light.py:
```diff
@@ -287,124 +268,113 @@ def _lights_setup_helper(klass):
-    @service_call
-    def service_set_mode(target_device, params):
-        target_device.set_mode(**params)
+async def _async_setup_services(hass: HomeAssistant):
```
This doesn't need to be a coroutine function since we don't await inside. Decorate it with `@callback`.

shenxn committed:
Change service setup to callback

shenxn committed:
Rename ip_addr to host

shenxn commented:
According to the suggestion from @rytilahti, I've changed all `ipaddr` to `host`. To maintain consistency, I also changed the key in config entries. If this PR is going to be merged after the release of Yeelight config flow (#37191), migration of config entries will be needed.

MartinHjelmare commented on homeassistant/components/yeelight/__init__.py:
```diff
@@ -472,7 +469,7 @@ def turn_off(self, duration=DEFAULT_TRANSITION, light_type=None):
         except BulbException as ex:
             _LOGGER.error(
-                "Unable to turn the bulb off: %s, %s: %s", self.ipaddr, self.name, ex
+                "Unable to turn the bulb off: %s, %s: %s", self.host, self.name, ex
```
There are four places in this class where we use `self.host` and three places where we use `self._host`. I suggest to be consistent and use `self._host` inside the class. Use the property for external access.
```suggestion
                "Unable to turn the bulb off: %s, %s: %s", self._host, self.name, ex
```

MartinHjelmare commented on homeassistant/components/yeelight/__init__.py:
```diff
@@ -486,7 +483,7 @@ def _update_properties(self):
             if self._available:  # just inform once
                 _LOGGER.error(
-                    "Unable to update device %s, %s: %s", self.ipaddr, self.name, ex
+                    "Unable to update device %s, %s: %s", self.host, self.name, ex
```
```suggestion
                    "Unable to update device %s, %s: %s", self._host, self.name, ex
```

MartinHjelmare commented on homeassistant/components/yeelight/__init__.py:
```diff
@@ -498,22 +495,22 @@ def _get_capabilities(self):
             _LOGGER.debug(
                 "Device %s, %s capabilities: %s",
-                self.ipaddr,
+                self.host,
```
```suggestion
                self._host,
```

MartinHjelmare commented on homeassistant/components/yeelight/__init__.py:
```diff
@@ -498,22 +495,22 @@ def _get_capabilities(self):
             _LOGGER.error(
                 "Unable to get device capabilities %s, %s: %s",
-                self.ipaddr,
+                self.host,
```
```suggestion
                self._host,
```

shenxn committed:
Use _host inside class

MartinHjelmare reviewed (approved):
Nice!
